src/common/src/util/value_encoding/mod.rs

```diff
@@ -86,14 +89,11 @@ fn serialize_value(value: ScalarRefImpl, mut buf: impl BufMut) {
         ScalarRefImpl::NaiveTime(v) => {
             serialize_naivetime(v.0.num_seconds_from_midnight(), v.0.nanosecond(), buf)
         }
-        ScalarRefImpl::Struct(StructRef::ValueRef { val }) => {
-            serialize_struct_or_list(val.to_protobuf_owned(), buf);
-        }
-        ScalarRefImpl::List(ListRef::ValueRef { val }) => {
-            serialize_struct_or_list(val.to_protobuf_owned(), buf);
+        ScalarRefImpl::Struct(s) => {
+            serialize_struct_or_list(s.to_protobuf_owned(), buf);
```

Comment:
Off-topic: the performance of serializing the protobuf representation is poor. We may improve it later. cc @Gun9niR